src/db/models/user.model.js

```diff
@@ -172,5 +264,40 @@ module.exports = (sequelize, DataTypes) => {
             });
     };
 
+    User.prototype.updateVerifyAccountToken = function updateVerifyAccountToken(email, expTime) {
+        return randomBytes(20)
+            .then((buf) => {
+                const token = buf.toString('hex');
+                return token;
+            }).then((token) => {
+                this.contactMethodVerificationToken = token;
+                this.contactMethodToVerify = email;
+                const m = moment.utc();
+                m.add(expTime, 'seconds');
+                this.contactMethodVerificationTokenExpires = m.format();
+                return this.save().then(() => token);
+            });
+    };
+
+    User.prototype.addVerifiedContact = function addVerifiedContact(contactMethodToVerify) {
+        if (this.verifiedContactMethods.includes(contactMethodToVerify)) {
```

Comment:
What is the use case for this entire block?

Reply:
prevents a user form re-verifying an email they've already verified

Reply:
it specifically dumps out the token and tells them it succeeded.